### PR TITLE
[fix-9991][worker]fix statement is closed before resultSet.getMetaData()

### LINKED ISSUE #9991: [Bug-BE] [sql component] sql component failed to run
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

sql component failed to run

image

### What you expected to happen

sql component runs successfully

### How to reproduce

[ERROR] 2022-05-11 09:31:17.553 +0000 [taskAppId=TASK-20220511-5469970732736_2-22-42] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.sql.SqlTask:[213] - execute sql error: Operation not allowed after ResultSet closed
[ERROR] 2022-05-11 09:31:17.553 +0000 [taskAppId=TASK-20220511-5469970732736_2-22-42] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.sql.SqlTask:[164] - sql task error: java.sql.SQLException: Operation not allowed after ResultSet closed

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by huangchenguang123**
fix statement is closed before resultSet.getMetaData(), fix #9991

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java [hunk 1]
```diff
@@ -197,7 +198,9 @@ public void executeFuncAndSql(List<SqlBinds> mainStatementsBinds,
             // decide whether to executeQuery or executeUpdate based on sqlType
             if (sqlParameters.getSqlType() == SqlType.QUERY.ordinal()) {
                 // query statements need to be convert to JsonArray and inserted into Alert to send
-                resultSet = executeQuery(connection, mainStatementsBinds.get(0), "main");
+                statement = prepareStatementAndBind(connection, mainStatementsBinds.get(0));
+                logger.info("main statement execute query, for sql: {}", mainStatementsBinds.get(0).getSql());
+                resultSet = statement.executeQuery();
```
Comment (zhuangchong) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java]: I do not recommend this modification. You can put the resultProcess method in the query method, and the close method only has connection as an input parameter.

Comment (huangchenguang123) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java]: Its better to put the resultProcess method in the query method, i will fix my code.

Comment (zhuangchong) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java]: the close method only has connection as an input parameter, it is also recommended to modify.

Comment (huangchenguang123) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java]: resultSet is allways null， remove from the close method

**Comment by huangchenguang123** [comment]
@SbloodyS when i push my code and trigger check, sometimes the E2E/FileManage will fail, but i think my code won't effect the test case, I would like to know why this is causing this problem, this is my first contribute, so could you tell me the reasion?

**Comment by SbloodyS** [comment]
> @SbloodyS when i push my code and trigger check, sometimes the E2E/FileManage will fail, but i think my code won't effect the test case, I would like to know why this is causing this problem, this is my first contribute, so could you tell me the reasion?

I had restart the failed tests.  You can try to trigger the retry by git push or closing and then reopening PR.

There may be some instability in E2E tests. I'll fix it in a few days.

**Review by zhuangchong — APPROVED** [review]
+1

**Comment by zhuangchong** [comment]
@huangchenguang123 
Thank you very much for your contribution, this PR is very good. I think we can get deep communiction, you can contact me through mail or add wechat(KerwinZhuang), when mail or added, please tell me who you are, I think I can help to familiar with the DolphinScheduler if you meet with problems.

**Pull request merged**